src/body.rs

```diff
@@ -150,6 +150,42 @@ impl<T: 'static + Send + serde::Serialize> IntoResponse for Json<T> {
     }
 }
 
+/// A wrapper for form encoded (application/x-www-form-urlencoded) (de)serialization of bodies.
+///
+/// This type is usable both as an extractor (argument to an endpoint) and as a response
+/// (return value from an endpoint), though returning a response with form data is uncommon
+/// and probably not good practice.
+pub struct Form<T>(pub T);
+
+impl<T: Send + serde::de::DeserializeOwned + 'static, S: 'static> Extract<S> for Form<T> {
+    // Note: cannot use `existential type` here due to ICE
+    type Fut = FutureObj<'static, Result<Self, Response>>;
+
+    fn extract(data: &mut S, req: &mut Request, params: &RouteMatch<'_>) -> Self::Fut {
+        let mut body = std::mem::replace(req.body_mut(), Body::empty());
+        FutureObj::new(Box::new(
+            async move {
+                let body = await!(body.read_to_vec()).map_err(mk_err)?;
+                let data: T = serde_qs::from_bytes(&body).map_err(mk_err)?;
+                Ok(Form(data))
+            },
+        ))
+    }
+}
+
+impl<T: 'static + Send + serde::Serialize> IntoResponse for Form<T> {
+    fn into_response(self) -> Response {
+        // TODO: think about how to handle errors
+        http::Response::builder()
+            .status(http::status::StatusCode::OK)
+            .header("Content-Type", "Application/x-www-form-urlencoded")
```

Comment:
It would be nice if the mime type is written in lowercase.

```suggestion
            .header("Content-Type", "application/x-www-form-urlencoded")
```

Reply:
Agree on this, it's [written this way](https://github.com/rust-net-web/tide/blob/73dfd4dc260eafcde5252eab736470302bf1a11c/src/body.rs#L147) for `json`, I didn't know why and figured there might be a reason for it, so went with the same capitalization.

Reply:
I think that was in turn copied from Rocket :joy:

Lower-casing is good, feel free to fix json  while you're at it.